bbq/vm/vm.go

```diff
@@ -657,69 +672,39 @@ func opInvokeMethodDynamic(vm *VM, ins opcode.InstructionInvokeMethodDynamic) {
 	// Load type arguments
 	typeArguments := loadTypeArguments(vm, ins.TypeArgs)
 
-	// Get receiver
-	argumentCount := int(ins.ArgCount)
-	receiver := getReceiver(vm, argumentCount)
-	memberAccessibleValue := receiver.(interpreter.MemberAccessibleValue)
+	explicitArgumentsCount := int(ins.ArgCount)
+	arguments := vm.peekN(explicitArgumentsCount)
+	receiver := arguments[receiverIndex]
+	arguments[receiverIndex] = unwrapReceiver(vm.context, receiver)
 
 	// Get function
 	nameIndex := ins.Name
 	funcName := getStringConstant(vm, nameIndex)
 
+	memberAccessibleValue := receiver.(interpreter.MemberAccessibleValue)
 	functionValue := memberAccessibleValue.GetMember(
 		vm.context,
 		EmptyLocationRange,
 		funcName,
-	)
+	).(*BoundFunctionPointerValue)
 
 	invokeFunction(
 		vm,
-		functionValue,
-		true,
+		functionValue.Method,
+		explicitArgumentsCount,
+		arguments,
 		typeArguments,
 	)
 }
 
 func invokeFunction(
 	vm *VM,
 	functionValue Value,
-	hasExplicitReceiver bool,
+	explicitArgumentsCount int,
+	arguments []Value,
 	typeArguments []bbq.StaticType,
 ) {
```

Comment:
Simplified this `invokeFunction` method, by passing the arguments rather than trying to figure out the arguments inside this function for different kinds of function-values/invocations. The invocation instruction is responsible for passing the correct arguments. Reduces the unnecessary checks.